hathor/p2p/manager.py

```diff
@@ -104,6 +107,12 @@ def __init__(self, reactor: ReactorBase, my_peer: PeerId, server_factory: 'Hatho
 
         self.ssl = ssl
 
+        # Object that handles the sync until the last checkpoint for all peers
+        self.sync_checkpoints = SyncCheckpoint(manager)
```

Comment:
I don't understand that suggestion. `ConnectionsManager` isn't a factory right? It's more of a factory holder. But the main point is that we don't know if we have to use sync-v1 or sync-v2 until after the handshake, so we can't easily have different protocol factories (currently we have different "plugins" for sync-v1 and sync-v2, they are not actually plugins anymore after a recent refactor). But either way the current `SyncCheckpoint` class works as a singleton and the most obvious place to put that singleton in is the `ConnectionsManager`. We could just not instantiate it when sync-v2 isn't enabled, but it doesn't make much difference, it will only have any runtime impact if it is started, an only sync-v2 starts it.

Reply:
I've removed `SyncCheckpoint` from this PR, we can discuss this on #275

Reply:
Ok. Great!